### DISCUSSION TITLE
Thonny says file not found even when the file is there.

### DISCUSSION THREAD

**Discussion by AppalachianWarrior** [micropython/micropython#15379] [RP2040 / Pico]
I am an Arduino/C++ guy and I am sorry if I use a different 'lingo' to explain the issue. 

I hit F5 to run this file named TestBed1.py. The Thonny IDE is saying that it can't find the **ble_advertising.py** header file.

In the image you can see that the ble_advertising.py file is indeed in the same folder as TestBed1.py

Why is the Thonny IDE saying this when the required file is indeed present in the same folder as the file that I am trying to execute?

![twe](https://github.com/micropython/micropython/assets/72459209/a738cafb-6e75-4f24-9ee7-926a6b291f4f)

**Comment by Josverl** [comment]
I think your question is better posted to the [Thonny discussions](https://github.com/thonny/thonny/discussions)

**Comment by beetlegigg** [comment]
I see your modules to be imported are in a sub directory.  Modules to be imported will be looked for in the top dir (i.e Raspberry Pi Pico) or in a sub dir called /lib.   To import your module sitting in a sub dir then use import in the form of mysubdir.mymodule.   Perhaps to make your life easier I suggest you consider not using such a verbose sub dir name :-)

**Reply by scruss** [reply]
Yeah, basically do that thing.

Thonny doesn't seem to execute files in the editor from what you might think is a current directory.  Put ble_advertising.py in /lib, and all should work better.
